### DISCUSSION TITLE
Could not resolve dependency: peer next-auth@"^4" from @next-auth/prisma-adapter@1.0.7

### DISCUSSION THREAD

**Discussion by happyrw** [vercel/next.js#62273] [Help]
### Summary

<html>
<body>

npm ERR! code ERESOLVE
12:36:33.408 | npm ERR! ERESOLVE could not resolve
12:36:33.409 | npm ERR!
12:36:33.409 | npm ERR! While resolving: @next-auth/prisma-adapter@1.0.7
12:36:33.409 | npm ERR! Found: next-auth@5.0.0-beta.5
12:36:33.409 | npm ERR! node_modules/next-auth
12:36:33.409 | npm ERR!   next-auth@"^5.0.0-beta.5" from the root project
12:36:33.409 | npm ERR!
12:36:33.409 | npm ERR! Could not resolve dependency:
12:36:33.409 | npm ERR! peer next-auth@"^4" from @next-auth/prisma-adapter@1.0.7
12:36:33.409 | npm ERR! node_modules/@next-auth/prisma-adapter
12:36:33.409 | npm ERR!   @next-auth/prisma-adapter@"^1.0.7" from the root project
12:36:33.409 | npm ERR!
12:36:33.409 | npm ERR! Conflicting peer dependency: next-auth@4.24.6
12:36:33.410 | npm ERR! node_modules/next-auth
12:36:33.410 | npm ERR!   peer next-auth@"^4" from @next-auth/prisma-adapter@1.0.7
12:36:33.410 | npm ERR!   node_modules/@next-auth/prisma-adapter
12:36:33.410 | npm ERR!     @next-auth/prisma-adapter@"^1.0.7" from the root project
12:36:33.410 | npm ERR!
12:36:33.410 | npm ERR! Fix the upstream dependency conflict, or retry
12:36:33.410 | npm ERR! this command with --force or --legacy-peer-deps
12:36:33.410 | npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
12:36:33.410 | npm ERR!
12:36:33.410 | npm ERR!
12:36:33.410 | npm ERR! For a full report see:
12:36:33.410 | npm ERR! /vercel/.npm/_logs/2024-02-20T10_36_31_983Z-eresolve-report.txt
12:36:33.411 |  
12:36:33.411 | npm ERR! A complete log of this run can be found in: /vercel/.npm/_logs/2024-02-20T10_36_31_983Z-debug-0.log
12:36:33.425 | Error: Command "npm install" exited with 1
12:36:33.678

<br class="Apple-interchange-newline">
</body>
</html>

### Additional information

```js
package.json

{
  "name": "auth_tutorial",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "postinstall": "prisma generate"
  },
  "dependencies": {
    "@auth/prisma-adapter": "^1.2.0",
    "@hookform/resolvers": "^3.3.4",
    "@next-auth/prisma-adapter": "^1.0.7",
    "@prisma/client": "^5.9.1",
    "bcrypt": "^5.1.1",
    "bcryptjs": "^2.4.3",
    "next": "14.1.0",
    "next-auth": "^5.0.0-beta.5",
    "react": "^18",
    "react-dom": "^18",
    "react-hook-form": "^7.50.0",
    "react-icons": "^5.0.1",
    "react-spinners": "^0.13.8",
    "react-toastify": "^10.0.4",
    "resend": "^3.1.0",
    "uuid": "^9.0.1",
    "yup": "^1.3.3"
  },
  "devDependencies": {
    "@types/bcryptjs": "^2.4.6",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "@types/uuid": "^9.0.8",
    "autoprefixer": "^10.0.1",
    "eslint": "^8",
    "eslint-config-next": "14.1.0",
    "postcss": "^8",
    "prisma": "^5.9.1",
    "tailwindcss": "^3.3.0",
    "typescript": "^5"
  }
}
```

**Comment by germanmeneses** [comment]
Could you solve it? I have exactly the error. I bypassed with `npm install xxx --legacy-peer-deps` but I don't know what is happening

**Comment by rmymartinet** [comment]
@happyrw

### 1. Problem Encountered

I had a similar issue where the Prisma Adapter wasn’t compatible with the latest beta version au auth.js and prisma adapter.

package version :      

- "next-auth": "^5.0.0-beta.25",
- "@prisma/client": "^6.2.1",
- "@auth/prisma-adapter": "^2.7.4",
- "@next-auth/prisma-adapter": "^1.0.7",
- "@prisma/client": "^6.2.1",

This caused an error, and I struggled to find a proper solution.
The previously suggested command (npm install xxx-legacy-peer-deps) didn’t work for me either.


### 2. How I Solved It

Here’s what I did to resolve the issue:
	1.	Updated the Middleware: I adjusted my middleware
![Capture d’écran, le 2025-01-22 à 12 58 28](https://github.com/user-attachments/assets/19c58439-2c89-47d4-ba7c-d2021e93aa53)

	2.	Added an .npmrc File in my root directory
	•	I created an .npmrc file in my project.
	•	Inside it, I added this line: legacy-peer-deps=true

By combining these two changes, I managed to fix the problem, and everything is now working as expected.

3. Conclusion

I hope this solution helps! If you need further clarification, feel free to ask.
